src/main/java/cs2103/v15_1j/jimjim/ui/TaskEventRowFactory.java

```diff
@@ -78,52 +80,51 @@ public void showAllTaskEvents(){
 					itemToAdd = nextDeadlineTask;
 					deadlineTaskCounter++;
 				}
-		    }
-		    
-		    LocalDate tempDate = getLatestDate(currentDate, itemToAdd);
-		    
-		    if(currentDate == null){
-		    	currentDate = tempDate;
-		    	addLabel(currentDate);
-		    }
-		    else if(!currentDate.equals(tempDate)){
-		    	currentDate = tempDate;
-		    	addLabel(currentDate);
-		    }
-		    
-		    addTaskEvent(itemToAdd);
+			}
+
+			LocalDate tempDate = getLatestDate(currentDate, itemToAdd);
```

Comment:
I totally don't understand `getLatestDate` and `tempDate`. Shouldn't you just update `currentDate` once you've chosen `itemToAdd` above? `currentDate = nextEvent/nextDeadlineTask .getDate`